New OSGi Code Error

Rather than continue with my battle with V14 compatibility mode, over the weekend I decided to spend the time to
 investigate the new OSGi code.

The approach appears to be a significant improvement on the previous instantiator etc..

Although the new base-starter compiles perfectly, when trying to run it I get the error below.

I have already started the  process to create a Karaf "feature module" but this error is preventing further progress in this task.

Realizing that it is very early in the implementation and testing of the new OSGi approach I would like to be involved at this stage, to help Vaadin, where ever I can, to achieve a very successful outcome for  the new OSGi approach.

Especially using it with Karaf.

For many years I had been totally convinced that bnd and bndtools was the the only "pure" way to use OSGi.

Since changing to Karaf my project has advanced at a much greater pace than with bnd and bndtools.

Paul Fraser

  ```
paulf@paulf-X550LD:~$ cd IdeaProjects/base-starter-flow-osgi/
  paulf@paulf-X550LD:~/IdeaProjects/base-starter-flow-osgi$ mvn -pl app bnd-run:run
  WARNING: An illegal reflective access operation has occurred
  WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
  WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
  WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
  WARNING: All illegal access operations will be denied in a future release
  [INFO] Scanning for projects...
  [WARNING] 
  [WARNING] Some problems were encountered while building the effective model for com.example:application:pom:1.0-SNAPSHOT
  [WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.servicemix.specs:org.apache.servicemix.specs.jaxb-api-2.2:jar -> duplicate declaration of version 2.8.0 @ line 109, column 21
  [WARNING] 
  [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
  [WARNING] 
  [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
  [WARNING] 
  [INFO] 
  [INFO] ----------------------< com.example:application >-----------------------
  [INFO] Building The Aplication that runs Vaadin Flow and in OSGi 1.0-SNAPSHOT
  [INFO] --------------------------------[ pom ]---------------------------------
  [INFO] 
  [INFO] --- bnd-run-maven-plugin:5.2.0:run (default-cli) @ application ---
  [WARNING] The POM for com.vaadin:flow-server-compatibility-mode:jar:6.0-SNAPSHOT is missing, no dependency information available
  [WARNING] Could not transfer metadata com.vaadin:flow-bom:4.0-SNAPSHOT/maven-metadata.xml from/to ${mpr.prerelease.repo.id} (${mpr.prerelease.repo.url}): Cannot access ${mpr.prerelease.repo.url} with type default using the available connector factories: BasicRepositoryConnectorFactory
  [WARNING] Could not transfer metadata com.vaadin:flow-bom:4.0-SNAPSHOT/maven-metadata.xml from/to ${mpr.addons.repo.id} (${mpr.addons.repo.url}): Cannot access ${mpr.addons.repo.url} with type default using the available connector factories: BasicRepositoryConnectorFactory
  [WARNING] Could not transfer metadata com.vaadin:flow-project:4.0-SNAPSHOT/maven-metadata.xml from/to ${mpr.prerelease.repo.id} (${mpr.prerelease.repo.url}): Cannot access ${mpr.prerelease.repo.url} with type default using the available connector factories: BasicRepositoryConnectorFactory
  [WARNING] Could not transfer metadata com.vaadin:flow-project:4.0-SNAPSHOT/maven-metadata.xml from/to ${mpr.addons.repo.id} (${mpr.addons.repo.url}): Cannot access ${mpr.addons.repo.url} with type default using the available connector factories: BasicRepositoryConnectorFactory
  [WARNING] Failed to resolve dependency com.vaadin:flow:jar:6.0-SNAPSHOT
  [WARNING] Failed to resolve dependency com.vaadin:flow-server-compatibility-mode:jar:6.0-SNAPSHOT
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key swagger-parser-v2-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key swagger-parser-v2-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key commons-lang-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-core:2.0.5] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key jmustache-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key commons-lang-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key commons-cli-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key reflections-version
  [INFO] Undefined property in POM[com.vaadin:mpr-v8:4.0-SNAPSHOT] : key framework.8.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v8:4.0-SNAPSHOT] : key framework.8.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v8:4.0-SNAPSHOT] : key framework.8.version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-parser-version-v1
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key jmustache-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key logback-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key logback-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key commons-lang-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key commons-cli-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key reflections-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-codegen-generators-version
  [INFO] Undefined property in POM[com.vaadin:mpr-core:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[com.vaadin:mpr-core:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[com.vaadin:mpr-core:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key testng-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key jmockit-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key project.parent.groupId
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key junit-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key wiremock-version
  [INFO] Undefined property in POM[com.vaadin:vaadin-spring:16.0-SNAPSHOT] : key vaadin.flow.version
  [INFO] Undefined property in POM[io.swagger:swagger-compat-spec-parser:1.0.33] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-compat-spec-parser:1.0.33] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger:swagger-compat-spec-parser:1.0.33] : key junit-version
  [INFO] Undefined property in POM[com.google.errorprone:error_prone_annotations:2.2.0] : key junit.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v7:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v7:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v7:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key jmockit-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key junit-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key commons-lang-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key wiremock-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key jmockit-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key jackson-version
  WARNING: An illegal reflective access operation has occurred
  WARNING: Illegal reflective access by org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender (file:/home/paulf/.m2/repository/org/apache/felix/org.apache.felix.framework/6.0.3/org.apache.felix.framework-6.0.3.jar) to method java.net.URLClassLoader.addURL(java.net.URL)
  WARNING: Please consider reporting this to the maintainers of org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender
  WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
  WARNING: All illegal access operations will be denied in a future release
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.equals.DefaultEqualsImplementationRegistrarSPI of service com.helger.commons.equals.IEqualsImplementationRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.hashcode.DefaultHashCodeImplementationRegistrarSPI of service com.helger.commons.hashcode.IHashCodeImplementationRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.serialize.convert.BasicSerializationConverterRegistrar of service com.helger.commons.serialize.convert.ISerializationConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.thirdparty.ThirdPartyModuleProvider_ph_commons of service com.helger.commons.thirdparty.IThirdPartyModuleProviderSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.BaseTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.CharsetTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.CollectionTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.DateTimeTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.IOTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.LocaleTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.MutableTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.equals.DefaultEqualsImplementationRegistrarSPI of service com.helger.commons.equals.IEqualsImplementationRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.hashcode.DefaultHashCodeImplementationRegistrarSPI of service com.helger.commons.hashcode.IHashCodeImplementationRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.serialize.convert.BasicSerializationConverterRegistrar of service com.helger.commons.serialize.convert.ISerializationConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.thirdparty.ThirdPartyModuleProvider_ph_commons of service com.helger.commons.thirdparty.IThirdPartyModuleProviderSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.BaseTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.CharsetTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.CollectionTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.DateTimeTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.IOTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.LocaleTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.MutableTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider org.eclipse.jetty.http.Http1FieldPreEncoder of service org.eclipse.jetty.http.HttpFieldPreEncoder in bundle org.apache.felix.http.jetty
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider org.eclipse.jetty.http2.hpack.HpackFieldPreEncoder of service org.eclipse.jetty.http.HttpFieldPreEncoder in bundle org.apache.felix.http.jetty
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider org.eclipse.jetty.http.Http1FieldPreEncoder of service org.eclipse.jetty.http.HttpFieldPreEncoder in bundle org.apache.felix.http.jetty
  Jan. 04, 2021 3:21:30 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider org.eclipse.jetty.http2.hpack.HpackFieldPreEncoder of service org.eclipse.jetty.http.HttpFieldPreEncoder in bundle org.apache.felix.http.jetty
  [main] INFO org.eclipse.jetty.util.log - Logging initialized @1082ms to org.eclipse.jetty.util.log.Slf4jLog
  [main] INFO org.eclipse.jetty.server.Server - jetty-9.4.31.v20200723; built: 2020-07-23T17:57:36.812Z; git: 450ba27947e13e66baa8cd1ce7e85a4461cacc1d; jvm 11.0.9.1+1-Ubuntu-0ubuntu1.20.04
  [main] INFO org.eclipse.jetty.server.session - DefaultSessionIdManager workerName=node0
  [main] INFO org.eclipse.jetty.server.session - No SessionScavenger set, using defaults
  [main] INFO org.eclipse.jetty.server.session - node0 Scavenging every 600000ms
  [main] INFO org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@21337f7b{/,null,AVAILABLE}
  [main] INFO org.eclipse.jetty.server.Server - Started @1349ms
  [main] INFO org.eclipse.jetty.server.session - node0 Scavenging every 660000ms
  [ERROR] Failed to start Connector: ServerConnector@31000e60{HTTP/1.1, (http/1.1)}{localhost:8080}
  java.io.IOException: Failed to bind to localhost/127.0.0.1:8080
	  at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:349)
	  at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:310)
	  at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80)
	  at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:234)
	  at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	  at org.apache.felix.http.jetty.internal.JettyService.startConnector(JettyService.java:619)
	  at org.apache.felix.http.jetty.internal.JettyService.initializeHttp(JettyService.java:449)
	  at org.apache.felix.http.jetty.internal.JettyService.initializeJetty(JettyService.java:347)
	  at org.apache.felix.http.jetty.internal.JettyService.startJetty(JettyService.java:192)
	  at org.apache.felix.http.jetty.internal.JettyService.start(JettyService.java:115)
	  at org.apache.felix.http.jetty.internal.JettyActivator.doStart(JettyActivator.java:66)
	  at org.apache.felix.http.base.internal.AbstractActivator.start(AbstractActivator.java:39)
	  at org.apache.felix.framework.util.SecureAction.startActivator(SecureAction.java:698)
	  at org.apache.felix.framework.Felix.activateBundle(Felix.java:2402)
	  at org.apache.felix.framework.Felix.startBundle(Felix.java:2308)
	  at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:998)
	  at aQute.launcher.Launcher.start(Launcher.java:674)
	  at aQute.launcher.Launcher.startBundles(Launcher.java:654)
	  at aQute.launcher.Launcher.activate(Launcher.java:560)
	  at aQute.launcher.Launcher.launch(Launcher.java:400)
	  at aQute.launcher.Launcher.run(Launcher.java:181)
	  at aQute.launcher.Launcher.main(Launcher.java:157)
	  at aQute.launcher.pre.EmbeddedLauncher.executeWithRunPath(EmbeddedLauncher.java:170)
	  at aQute.launcher.pre.EmbeddedLauncher.findAndExecute(EmbeddedLauncher.java:135)
	  at aQute.launcher.pre.EmbeddedLauncher.main(EmbeddedLauncher.java:52)
  Caused by: java.net.BindException: Address already in use
	  at java.base/sun.nio.ch.Net.bind0(Native Method)
	  at java.base/sun.nio.ch.Net.bind(Net.java:455)
	  at java.base/sun.nio.ch.Net.bind(Net.java:447)
	  at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	  at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:80)
	  at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:345)
	  ... 24 more
  [main] INFO org.eclipse.jetty.server.session - node0 Stopped scavenging
  [main] INFO org.eclipse.jetty.server.handler.ContextHandler - Stopped o.e.j.s.ServletContextHandler@21337f7b{/,null,UNAVAILABLE}
  [INFO] Stop[peEdR RJOeRt]t yJ.ett
  y stopped (no connectors available)
  [ERROR] Exception while initializing Jetty.
  java.lang.NullPointerException
	  at org.apache.felix.http.jetty.internal.JettyService.initializeJetty(JettyService.java:392)
	  at org.apache.felix.http.jetty.internal.JettyService.startJetty(JettyService.java:192)
	  at org.apache.felix.http.jetty.internal.JettyService.start(JettyService.java:115)
	  at org.apache.felix.http.jetty.internal.JettyActivator.doStart(JettyActivator.java:66)
	  at org.apache.felix.http.base.internal.AbstractActivator.start(AbstractActivator.java:39)
	  at org.apache.felix.framework.util.SecureAction.startActivator(SecureAction.java:698)
	  at org.apache.felix.framework.Felix.activateBundle(Felix.java:2402)
	  at org.apache.felix.framework.Felix.startBundle(Felix.java:2308)
	  at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:998)
	  at aQute.launcher.Launcher.start(Launcher.java:674)
	  at aQute.launcher.Launcher.startBundles(Launcher.java:654)
	  at aQute.launcher.Launcher.activate(Launcher.java:560)
	  at aQute.launcher.Launcher.launch(Launcher.java:400)
	  at aQute.launcher.Launcher.run(Launcher.java:181)
	  at aQute.launcher.Launcher.main(Launcher.java:157)
	  at aQute.launcher.pre.EmbeddedLauncher.executeWithRunPath(EmbeddedLauncher.java:170)
	  at aQute.launcher.pre.EmbeddedLauncher.findAndExecute(EmbeddedLauncher.java:135)
	  at aQute.launcher.pre.EmbeddedLauncher.main(EmbeddedLauncher.java:52)
  [main] INFO com.vaadin.flow.osgi.support.ServletContainerInitializerClasses - Unable to detect used OSGi framework version due to null
  ^Cpaulf@paulf-X550LD:~/IdeaProjects/base-starter-flow-osgi$ mvn -pl app bnd-run:run
  WARNING: An illegal reflective access operation has occurred
  WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
  WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
  WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
  WARNING: All illegal access operations will be denied in a future release
  [INFO] Scanning for projects...
  [WARNING] 
  [WARNING] Some problems were encountered while building the effective model for com.example:application:pom:1.0-SNAPSHOT
  [WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.apache.servicemix.specs:org.apache.servicemix.specs.jaxb-api-2.2:jar -> duplicate declaration of version 2.8.0 @ line 109, column 21
  [WARNING] 
  [WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
  [WARNING] 
  [WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
  [WARNING] 
  [INFO] 
  [INFO] ----------------------< com.example:application >-----------------------
  [INFO] Building The Aplication that runs Vaadin Flow and in OSGi 1.0-SNAPSHOT
  [INFO] --------------------------------[ pom ]---------------------------------
  [INFO] 
  [INFO] --- bnd-run-maven-plugin:5.2.0:run (default-cli) @ application ---
  [WARNING] The POM for com.vaadin:flow-server-compatibility-mode:jar:6.0-SNAPSHOT is missing, no dependency information available
  [WARNING] Could not transfer metadata com.vaadin:flow-bom:4.0-SNAPSHOT/maven-metadata.xml from/to ${mpr.prerelease.repo.id} (${mpr.prerelease.repo.url}): Cannot access ${mpr.prerelease.repo.url} with type default using the available connector factories: BasicRepositoryConnectorFactory
  [WARNING] Could not transfer metadata com.vaadin:flow-bom:4.0-SNAPSHOT/maven-metadata.xml from/to ${mpr.addons.repo.id} (${mpr.addons.repo.url}): Cannot access ${mpr.addons.repo.url} with type default using the available connector factories: BasicRepositoryConnectorFactory
  [WARNING] Could not transfer metadata com.vaadin:flow-project:4.0-SNAPSHOT/maven-metadata.xml from/to ${mpr.prerelease.repo.id} (${mpr.prerelease.repo.url}): Cannot access ${mpr.prerelease.repo.url} with type default using the available connector factories: BasicRepositoryConnectorFactory
  [WARNING] Could not transfer metadata com.vaadin:flow-project:4.0-SNAPSHOT/maven-metadata.xml from/to ${mpr.addons.repo.id} (${mpr.addons.repo.url}): Cannot access ${mpr.addons.repo.url} with type default using the available connector factories: BasicRepositoryConnectorFactory
  [WARNING] Failed to resolve dependency com.vaadin:flow:jar:6.0-SNAPSHOT
  [WARNING] Failed to resolve dependency com.vaadin:flow-server-compatibility-mode:jar:6.0-SNAPSHOT
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key swagger-parser-v2-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key swagger-parser-v2-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key commons-lang-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v2-converter:2.0.5] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-core:2.0.5] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key jmustache-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key commons-lang-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key commons-cli-version
  [INFO] Undefined property in POM[io.swagger:swagger-codegen:2.3.1] : key reflections-version
  [INFO] Undefined property in POM[com.vaadin:mpr-v8:4.0-SNAPSHOT] : key framework.8.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v8:4.0-SNAPSHOT] : key framework.8.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v8:4.0-SNAPSHOT] : key framework.8.version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-parser-version-v1
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-parser-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key jmustache-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key logback-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key logback-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key commons-lang-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key commons-cli-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key reflections-version
  [INFO] Undefined property in POM[io.swagger.codegen.v3:swagger-codegen:3.0.2] : key swagger-codegen-generators-version
  [INFO] Undefined property in POM[com.vaadin:mpr-core:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[com.vaadin:mpr-core:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[com.vaadin:mpr-core:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key testng-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key jmockit-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key project.parent.groupId
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key junit-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger:swagger-parser:1.0.39] : key wiremock-version
  [INFO] Undefined property in POM[com.vaadin:vaadin-spring:16.0-SNAPSHOT] : key vaadin.flow.version
  [INFO] Undefined property in POM[io.swagger:swagger-compat-spec-parser:1.0.33] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger:swagger-compat-spec-parser:1.0.33] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger:swagger-compat-spec-parser:1.0.33] : key junit-version
  [INFO] Undefined property in POM[com.google.errorprone:error_prone_annotations:2.2.0] : key junit.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v7:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v7:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[com.vaadin:mpr-v7:4.0-SNAPSHOT] : key framework.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key jmockit-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key junit-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key commons-lang-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key slf4j-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser:2.0.5] : key wiremock-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key swagger-core-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key project.parent.version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key jmockit-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key commons-io-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key jackson-version
  [INFO] Undefined property in POM[io.swagger.parser.v3:swagger-parser-v3:2.0.5] : key jackson-version
  WARNING: An illegal reflective access operation has occurred
  WARNING: Illegal reflective access by org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender (file:/home/paulf/.m2/repository/org/apache/felix/org.apache.felix.framework/6.0.3/org.apache.felix.framework-6.0.3.jar) to method java.net.URLClassLoader.addURL(java.net.URL)
  WARNING: Please consider reporting this to the maintainers of org.apache.felix.framework.ext.ClassPathExtenderFactory$DefaultClassLoaderExtender
  WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
  WARNING: All illegal access operations will be denied in a future release
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.equals.DefaultEqualsImplementationRegistrarSPI of service com.helger.commons.equals.IEqualsImplementationRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.hashcode.DefaultHashCodeImplementationRegistrarSPI of service com.helger.commons.hashcode.IHashCodeImplementationRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.serialize.convert.BasicSerializationConverterRegistrar of service com.helger.commons.serialize.convert.ISerializationConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.thirdparty.ThirdPartyModuleProvider_ph_commons of service com.helger.commons.thirdparty.IThirdPartyModuleProviderSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.BaseTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.CharsetTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.CollectionTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.DateTimeTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.IOTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.LocaleTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:05 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.MutableTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.equals.DefaultEqualsImplementationRegistrarSPI of service com.helger.commons.equals.IEqualsImplementationRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.hashcode.DefaultHashCodeImplementationRegistrarSPI of service com.helger.commons.hashcode.IHashCodeImplementationRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.serialize.convert.BasicSerializationConverterRegistrar of service com.helger.commons.serialize.convert.ISerializationConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.thirdparty.ThirdPartyModuleProvider_ph_commons of service com.helger.commons.thirdparty.IThirdPartyModuleProviderSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.BaseTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.CharsetTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.CollectionTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.DateTimeTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.IOTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.LocaleTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider com.helger.commons.typeconvert.MutableTypeConverterRegistrar of service com.helger.commons.typeconvert.ITypeConverterRegistrarSPI in bundle com.helger.ph-commons
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider org.eclipse.jetty.http.Http1FieldPreEncoder of service org.eclipse.jetty.http.HttpFieldPreEncoder in bundle org.apache.felix.http.jetty
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider org.eclipse.jetty.http2.hpack.HpackFieldPreEncoder of service org.eclipse.jetty.http.HttpFieldPreEncoder in bundle org.apache.felix.http.jetty
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider org.eclipse.jetty.http.Http1FieldPreEncoder of service org.eclipse.jetty.http.HttpFieldPreEncoder in bundle org.apache.felix.http.jetty
  Jan. 04, 2021 3:24:06 PM org.apache.aries.spifly.BaseActivator log
  INFO: Registered provider org.eclipse.jetty.http2.hpack.HpackFieldPreEncoder of service org.eclipse.jetty.http.HttpFieldPreEncoder in bundle org.apache.felix.http.jetty
  [main] INFO org.eclipse.jetty.util.log - Logging initialized @1071ms to org.eclipse.jetty.util.log.Slf4jLog
  [main] INFO org.eclipse.jetty.server.Server - jetty-9.4.31.v20200723; built: 2020-07-23T17:57:36.812Z; git: 450ba27947e13e66baa8cd1ce7e85a4461cacc1d; jvm 11.0.9.1+1-Ubuntu-0ubuntu1.20.04
  [main] INFO org.eclipse.jetty.server.session - DefaultSessionIdManager workerName=node0
  [main] INFO org.eclipse.jetty.server.session - No SessionScavenger set, using defaults
  [main] INFO org.eclipse.jetty.server.session - node0 Scavenging every 600000ms
  [main] INFO org.eclipse.jetty.server.handler.ContextHandler - Started o.e.j.s.ServletContextHandler@21337f7b{/,null,AVAILABLE}
  [main] INFO org.eclipse.jetty.server.Server - Started @1416ms
  [main] INFO org.eclipse.jetty.server.session - node0 Scavenging every 600000ms
  [main] INFO org.eclipse.jetty.server.AbstractConnector - Started ServerConnector@31000e60{HTTP/1.1, (http/1.1)}{localhost:8080}
  [INFO] Started Jetty 9.4.31.v20200723 at port(s) HTTP:8080 on context path / [minThreads=8,maxThreads=200,acceptors=1,selectors=2]
  [main] INFO com.vaadin.flow.osgi.support.ServletContainerInitializerClasses - Unable to detect used OSGi framework version due to null
  [ERROR] [ServiceReference 44 from bundle 17 : com.vaadin.flow.osgi:6.0.0.202101030040 ref=[com.vaadin.flow.server.VaadinServiceInitListener, javax.servlet.http.HttpSessionListener, javax.servlet.ServletContextListener] properties={component.id=3, component.name=com.vaadin.flow.osgi.support.OSGiVaadinInitialization, objectClass=[com.vaadin.flow.server.VaadinServiceInitListener, javax.servlet.http.HttpSessionListener, javax.servlet.ServletContextListener], osgi.http.whiteboard.context.select=(&(osgi.http.whiteboard.context.name=*) (!(osgi.http.whiteboard.context.name=vaadinResourcesContext.*))), osgi.http.whiteboard.listener=true, service.bundleid=17, service.id=44, service.scope=bundle}] Exception while calling servlet context listener.
  java.lang.NullPointerException
	  at com.vaadin.flow.server.startup.ApplicationConfiguration.lambda$get$0(ApplicationConfiguration.java:53)
	  at com.vaadin.flow.server.VaadinServletContext.getAttribute(VaadinServletContext.java:73)
	  at com.vaadin.flow.server.startup.ApplicationConfiguration.get(ApplicationConfiguration.java:49)
	  at com.vaadin.flow.server.DeploymentConfigurationFactory.readBuildInfo(DeploymentConfigurationFactory.java:131)
	  at com.vaadin.flow.server.DeploymentConfigurationFactory.createInitParameters(DeploymentConfigurationFactory.java:110)
	  at com.vaadin.flow.server.VaadinServlet.createDeploymentConfiguration(VaadinServlet.java:185)
	  at com.vaadin.flow.server.VaadinServlet.createServletService(VaadinServlet.java:224)
	  at com.vaadin.flow.server.VaadinServlet.init(VaadinServlet.java:114)
	  at com.vaadin.flow.osgi.support.OSGiVaadinInitialization.initContext(OSGiVaadinInitialization.java:305)
	  at com.vaadin.flow.osgi.support.OSGiVaadinInitialization.contextInitialized(OSGiVaadinInitialization.java:244)
	  at org.apache.felix.http.base.internal.registry.EventListenerRegistry.contextInitialized(EventListenerRegistry.java:452)
	  at org.apache.felix.http.base.internal.whiteboard.WhiteboardManager.addWhiteboardService(WhiteboardManager.java:626)
	  at org.apache.felix.http.base.internal.whiteboard.tracker.WhiteboardServiceTracker.added(WhiteboardServiceTracker.java:104)
	  at org.apache.felix.http.base.internal.whiteboard.tracker.WhiteboardServiceTracker.addingService(WhiteboardServiceTracker.java:79)
	  at org.apache.felix.http.base.internal.whiteboard.tracker.WhiteboardServiceTracker.addingService(WhiteboardServiceTracker.java:36)
	  at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:943)
	  at org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:871)
	  at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	  at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229)
	  at org.osgi.util.tracker.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:903)
	  at org.apache.felix.framework.EventDispatcher.invokeServiceListenerCallback(EventDispatcher.java:990)
	  at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:838)
	  at org.apache.felix.framework.EventDispatcher.fireServiceEvent(EventDispatcher.java:545)
	  at org.apache.felix.framework.Felix.fireServiceEvent(Felix.java:4833)
	  at org.apache.felix.framework.Felix.registerService(Felix.java:3804)
	  at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:328)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:929)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:915)
	  at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:133)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager.registerService(AbstractComponentManager.java:984)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:752)
	  at org.apache.felix.scr.impl.manager.DependencyManager$SingleStaticCustomizer.addedService(DependencyManager.java:1271)
	  at org.apache.felix.scr.impl.manager.DependencyManager$SingleStaticCustomizer.addedService(DependencyManager.java:1222)
	  at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1200)
	  at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerAdded(ServiceTracker.java:1121)
	  at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.trackAdding(ServiceTracker.java:928)
	  at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.track(ServiceTracker.java:864)
	  at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1152)
	  at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:114)
	  at org.apache.felix.framework.EventDispatcher.invokeServiceListenerCallback(EventDispatcher.java:990)
	  at org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:838)
	  at org.apache.felix.framework.EventDispatcher.fireServiceEvent(EventDispatcher.java:545)
	  at org.apache.felix.framework.Felix.fireServiceEvent(Felix.java:4833)
	  at org.apache.felix.framework.Felix.registerService(Felix.java:3804)
	  at org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:328)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:929)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.register(AbstractComponentManager.java:915)
	  at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:133)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager.registerService(AbstractComponentManager.java:984)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager.activateInternal(AbstractComponentManager.java:752)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager.enableInternal(AbstractComponentManager.java:674)
	  at org.apache.felix.scr.impl.manager.AbstractComponentManager.enable(AbstractComponentManager.java:437)
	  at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.enableComponents(ConfigurableComponentHolder.java:667)
	  at org.apache.felix.scr.impl.BundleComponentActivator.initialEnable(BundleComponentActivator.java:305)
	  at org.apache.felix.scr.impl.Activator.loadComponents(Activator.java:554)
	  at org.apache.felix.scr.impl.Activator.access$200(Activator.java:70)
	  at org.apache.felix.scr.impl.Activator$ScrExtension.start(Activator.java:421)
	  at org.apache.felix.scr.impl.AbstractExtender.createExtension(AbstractExtender.java:196)
	  at org.apache.felix.scr.impl.AbstractExtender.modifiedBundle(AbstractExtender.java:169)
	  at org.apache.felix.scr.impl.AbstractExtender.addingBundle(AbstractExtender.java:139)
	  at org.apache.felix.scr.impl.AbstractExtender.addingBundle(AbstractExtender.java:49)
	  at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:475)
	  at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:420)
	  at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	  at org.osgi.util.tracker.AbstractTracked.trackInitial(AbstractTracked.java:183)
	  at org.osgi.util.tracker.BundleTracker.open(BundleTracker.java:159)
	  at org.apache.felix.scr.impl.AbstractExtender.startTracking(AbstractExtender.java:100)
	  at org.apache.felix.scr.impl.AbstractExtender.doStart(AbstractExtender.java:92)
	  at org.apache.felix.scr.impl.Activator.doStart(Activator.java:197)
	  at org.apache.felix.scr.impl.AbstractExtender.start(AbstractExtender.java:72)
	  at org.apache.felix.scr.impl.Activator.restart(Activator.java:164)
	  at org.apache.felix.scr.impl.config.ScrConfigurationImpl.configure(ScrConfigurationImpl.java:228)
	  at org.apache.felix.scr.impl.config.ScrConfigurationImpl.start(ScrConfigurationImpl.java:123)
	  at org.apache.felix.scr.impl.Activator.start(Activator.java:121)
	  at org.apache.felix.framework.util.SecureAction.startActivator(SecureAction.java:698)
	  at org.apache.felix.framework.Felix.activateBundle(Felix.java:2402)
	  at org.apache.felix.framework.Felix.startBundle(Felix.java:2308)
	  at org.apache.felix.framework.BundleImpl.start(BundleImpl.java:998)
	  at aQute.launcher.Launcher.start(Launcher.java:674)
	  at aQute.launcher.Launcher.startBundles(Launcher.java:654)
	  at aQute.launcher.Launcher.activate(Launcher.java:560)
	  at aQute.launcher.Launcher.launch(Launcher.java:400)
	  at aQute.launcher.Launcher.run(Launcher.java:181)
	  at aQute.launcher.Launcher.main(Launcher.java:157)
	  at aQute.launcher.pre.EmbeddedLauncher.executeWithRunPath(EmbeddedLauncher.java:170)
	  at aQute.launcher.pre.EmbeddedLauncher.findAndExecute(EmbeddedLauncher.java:135)
	  at aQute.launcher.pre.EmbeddedLauncher.main(EmbeddedLauncher.java:52)
  
  

```

--- a/drivers.xml
+++ b/drivers.xml
@@ -2,10 +2,10 @@
 <root>
     <windows>
         <driver id="googlechrome">
-            <version id="86.0.4240.22">
+            <version id="87.0.4280.88">
                 <bitrate thirtytwobit="true" sixtyfourbit="true">
-                    <filelocation>https://chromedriver.storage.googleapis.com/86.0.4240.22/chromedriver_win32.zip</filelocation>
-                    <hash>5bc302d49dd4afdbc538f026e56bc1c1ddb9d1c8</hash>
+                    <filelocation>https://chromedriver.storage.googleapis.com/87.0.4280.88/chromedriver_win32.zip</filelocation>
+                    <hash>f0bbea2cb52e7edede555a197e480b720fdf43bb</hash>
                     <hashtype>sha1</hashtype>
                 </bitrate>
             </version>
@@ -13,10 +13,10 @@
     </windows>
     <linux>
         <driver id="googlechrome">
-            <version id="86.0.4240.22">
+            <version id="87.0.4280.88">
                 <bitrate sixtyfourbit="true">
-                    <filelocation>https://chromedriver.storage.googleapis.com/86.0.4240.22/chromedriver_linux64.zip</filelocation>
-                    <hash>b8b531e4560e6c749daecbceb7584c5f0e6ae4da</hash>
+                    <filelocation>https://chromedriver.storage.googleapis.com/87.0.4280.88/chromedriver_linux64.zip</filelocation>
+                    <hash>06cf6155ad402e2a118943134d9c806d5b3fd6ec</hash>
                     <hashtype>sha1</hashtype>
                 </bitrate>
             </version>
@@ -24,14 +24,13 @@
     </linux>
     <osx>
         <driver id="googlechrome">
-            <version id="86.0.4240.22">
+            <version id="87.0.4280.88">
                 <bitrate sixtyfourbit="true">
-                    <filelocation>https://chromedriver.storage.googleapis.com/86.0.4240.22/chromedriver_mac64.zip</filelocation>
-                    <hash>3cbbd63803df46cea1cbc1eb5fb753de69248279</hash>
+                    <filelocation>https://chromedriver.storage.googleapis.com/87.0.4280.88/chromedriver_mac64.zip</filelocation>
+                    <hash>8f04b95747457b8a1e0800ccec7e3c7ca0e2c1fb</hash>
                     <hashtype>sha1</hashtype>
                 </bitrate>
             </version>
         </driver>
     </osx>
 </root>
-

--- a/flow-osgi/src/main/java/com/vaadin/flow/osgi/support/OSGiVaadinInitialization.java
+++ b/flow-osgi/src/main/java/com/vaadin/flow/osgi/support/OSGiVaadinInitialization.java
@@ -41,6 +41,7 @@ import org.osgi.service.component.annotations.Reference;
 import org.osgi.service.component.annotations.ServiceScope;
 import org.osgi.service.http.context.ServletContextHelper;
 import org.osgi.service.http.whiteboard.HttpWhiteboardConstants;
+import org.osgi.util.tracker.ServiceTracker;
 import org.slf4j.LoggerFactory;
 
 import com.vaadin.flow.di.Lookup;
@@ -52,6 +53,7 @@ import com.vaadin.flow.server.VaadinService;
 import com.vaadin.flow.server.VaadinServiceInitListener;
 import com.vaadin.flow.server.VaadinServlet;
 import com.vaadin.flow.server.VaadinServletContext;
+import com.vaadin.flow.server.startup.ApplicationConfigurationFactory;
 
 /**
  * Initialize {@link Lookup} and register internal resources (client engine
@@ -125,10 +127,10 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
 
         private final Bundle webAppBundle;
 
-        private OsgiLookupImpl(VaadinServletContext context)
-                throws IllegalContextState {
+        private OsgiLookupImpl(Bundle webAppBundle,
+                VaadinServletContext context) {
+            this.webAppBundle = webAppBundle;
             this.context = context;
-            webAppBundle = findBundle();
         }
 
         @Override
@@ -153,24 +155,119 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
             return webAppBundle;
         }
 
-        private Bundle findBundle() throws IllegalContextState {
-            ApplicationClassLoaderAccess classLoaderAccess = context
-                    .getAttribute(ApplicationClassLoaderAccess.class);
-            if (classLoaderAccess == null) {
-                throw new IllegalContextState(
-                        ApplicationClassLoaderAccess.class.getName()
-                                + "' instance is not available in "
-                                + VaadinContext.class.getName());
-            }
-            ClassLoader classloader = classLoaderAccess.getClassloader();
+    }
 
-            if (classloader instanceof BundleReference) {
-                return ((BundleReference) classloader).getBundle();
-            } else {
-                throw new IllegalStateException(
-                        "Unexpected classloader for the web app '" + classloader
-                                + "'. It's not possible to get an OSGi bundle from it");
+    private static class AppConfigFactoryTracker extends
+            ServiceTracker<ApplicationConfigurationFactory, ApplicationConfigurationFactory> {
+
+        private final Bundle webAppBundle;
+
+        private final VaadinServletContext context;
+
+        private final ServletContainerInitializerClasses initializerClasses;
+
+        private AppConfigFactoryTracker(Bundle webAppBundle,
+                VaadinServletContext context,
+                ServletContainerInitializerClasses initializerClasses)
+                throws IllegalContextState {
+            super(webAppBundle.getBundleContext(),
+                    ApplicationConfigurationFactory.class, null);
+            this.webAppBundle = webAppBundle;
+            this.context = context;
+            this.initializerClasses = initializerClasses;
+        }
+
+        @Override
+        public ApplicationConfigurationFactory addingService(
+                ServiceReference<ApplicationConfigurationFactory> reference) {
+            ApplicationConfigurationFactory factory = super.addingService(
+                    reference);
+            AppConfigFactoryTracker tracker = context
+                    .getAttribute(AppConfigFactoryTracker.class);
+            if (tracker != null) {
+                tracker.close();
+                context.removeAttribute(AppConfigFactoryTracker.class);
             }
+            initializeLookup();
+            return factory;
+        }
+
+        private void initializeLookup() {
+            /*
+             * There are two cases:
+             * 
+             * - the servlet context is initialized before all of the Servlets
+             * or
+             * 
+             * - after at least one of the servlets with the same context.
+             * 
+             * SevletContext::getClassLoader doesn't return the container the
+             * web app classloader for the servlet context instance here (in the
+             * "contextInitialized" method). It returns the web app classloader
+             * for the servlet context which is passed to the Servlet::init
+             * method: in fact two servlet context instances are different even
+             * though they share the same attributes and values. So to be able
+             * to identify the servlet context (which is coming from Servlet)
+             * and the "servletContext" instance here the Lookupx instance is
+             * used which is set as an attribute and if the context
+             * "is the same" the attributes should be the same.
+             */
+
+            // at this point at least one of the Servlet::init method should be
+            // called and the classloader should be available
+
+            // ensure the lookup is set into the context
+            Lookup lookup = context.getAttribute(Lookup.class,
+                    this::createLookup);
+            if (lookup == null) {
+                throw new IllegalStateException(
+                        VaadinContextInitializer.class.getSimpleName()
+                                + " has been executed but there is no "
+                                + ApplicationClassLoaderAccess.class
+                                        .getSimpleName()
+                                + " instance available");
+            }
+
+            Collection<Servlet> servlets = lookupAll(
+                    FrameworkUtil.getBundle(OSGiVaadinInitialization.class),
+                    Servlet.class);
+            for (Servlet servlet : servlets) {
+                if (isUninitializedServlet(servlet)) {
+                    handleUninitializedServlet(lookup, servlet);
+                }
+            }
+
+            initializerClasses.addContext(context.getContext());
+        }
+
+        private void handleUninitializedServlet(Lookup lookup,
+                Servlet servlet) {
+            ServletContext ctx = servlet.getServletConfig().getServletContext();
+            if (new VaadinServletContext(ctx)
+                    .getAttribute(Lookup.class) != lookup) {
+                return;
+            }
+            try {
+                servlet.init(servlet.getServletConfig());
+            } catch (ServletException e) {
+                LoggerFactory.getLogger(OSGiVaadinInitialization.class).error(
+                        "Couldn't initialize {} {}",
+                        VaadinServlet.class.getSimpleName(),
+                        servlet.getClass().getName(), e);
+            }
+        }
+
+        private Lookup createLookup() {
+            return new OsgiLookupImpl(webAppBundle, context);
+        }
+
+        private boolean isUninitializedServlet(Object object) {
+            if (object instanceof VaadinServlet) {
+                VaadinServlet vaadinServlet = (VaadinServlet) object;
+                return vaadinServlet.getServletConfig() != null
+                        && vaadinServlet.getService() == null;
+            }
+            return false;
         }
 
     }
@@ -247,77 +344,33 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
 
     private void initContext(VaadinContext context) {
         context.removeAttribute(VaadinContextInitializer.class);
+
         VaadinServletContext servletContext = (VaadinServletContext) context;
-
-        /*
-         * There are two cases:
-         * 
-         * - the servlet context is initialized before all of the Servlets or
-         * 
-         * - after at least one of the servlets with the same context.
-         * 
-         * SevletContext::getClassLoader doesn't return the container the web
-         * app classloader for the servlet context instance here (in the
-         * "contextInitialized" method). It returns the web app classloader for
-         * the servlet context which is passed to the Servlet::init method: in
-         * fact two servlet context instances are different even though they
-         * share the same attributes and values. So to be able to identify the
-         * servlet context (which is coming from Servlet) and the
-         * "servletContext" instance here the Lookupx instance is used which is
-         * set as an attribute and if the context "is the same" the attributes
-         * should be the same.
-         */
-
-        // at this point at least one of the Servlet::init method should be
-        // called and the classloader should be available
-
-        // ensure the lookup is set into the context
-
-        Lookup lookup = context.getAttribute(Lookup.class, () -> {
-            try {
-                return new OsgiLookupImpl(servletContext);
-            } catch (IllegalContextState exception) {
-                LoggerFactory.getLogger(OSGiVaadinInitialization.class)
-                        .warn("Couldn't create a lookup", exception);
-                return null;
-            }
-        });
-        if (lookup == null) {
-            throw new IllegalStateException(
-                    VaadinContextInitializer.class.getSimpleName()
-                            + " has been executed but there is no "
-                            + ApplicationClassLoaderAccess.class.getSimpleName()
-                            + " instance available");
+        try {
+            AppConfigFactoryTracker tracker = new AppConfigFactoryTracker(
+                    findBundle(servletContext), servletContext,
+                    initializerClasses);
+            servletContext.setAttribute(tracker);
+            tracker.open();
+        } catch (IllegalContextState exception) {
+            LoggerFactory.getLogger(OSGiVaadinInitialization.class)
+                    .warn("Couldn't initialize Vaadin Context", exception);
         }
-
-        Collection<Servlet> servlets = lookupAll(
-                FrameworkUtil.getBundle(OSGiVaadinInitialization.class),
-                Servlet.class);
-        for (Servlet servlet : servlets) {
-            if (isUninitializedServlet(servlet)) {
-                ServletContext ctx = servlet.getServletConfig()
-                        .getServletContext();
-                if (new VaadinServletContext(ctx)
-                        .getAttribute(Lookup.class) != lookup) {
-                    continue;
-                }
-                try {
-                    servlet.init(servlet.getServletConfig());
-                } catch (ServletException e) {
-                    LoggerFactory.getLogger(OSGiVaadinInitialization.class)
-                            .error("Couldn't initialize {} {}",
-                                    VaadinServlet.class.getSimpleName(),
-                                    servlet.getClass().getName(), e);
-                }
-            }
-        }
-
-        initializerClasses.addContext(servletContext.getContext());
     }
 
     @Override
     public void contextDestroyed(ServletContextEvent event) {
         initializerClasses.removeContext(event.getServletContext());
+
+        VaadinServletContext servletContext = (VaadinServletContext) event
+                .getServletContext();
+
+        AppConfigFactoryTracker tracker = servletContext
+                .getAttribute(AppConfigFactoryTracker.class);
+        if (tracker != null) {
+            tracker.close();
+            servletContext.removeAttribute(AppConfigFactoryTracker.class);
+        }
     }
 
     private void registerClientResources(String contextName) {
@@ -411,15 +464,6 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
         return builder.toString();
     }
 
-    private boolean isUninitializedServlet(Object object) {
-        if (object instanceof VaadinServlet) {
-            VaadinServlet vaadinServlet = (VaadinServlet) object;
-            return vaadinServlet.getServletConfig() != null
-                    && vaadinServlet.getService() == null;
-        }
-        return false;
-    }
-
     private static <T> Collection<T> lookupAll(Bundle bundle,
             Class<T> serviceClass) {
         try {
@@ -442,5 +486,26 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
         }
 
         return Collections.emptyList();
+    }
+
+    private Bundle findBundle(VaadinContext context)
+            throws IllegalContextState {
+        ApplicationClassLoaderAccess classLoaderAccess = context
+                .getAttribute(ApplicationClassLoaderAccess.class);
+        if (classLoaderAccess == null) {
+            throw new IllegalContextState(
+                    ApplicationClassLoaderAccess.class.getName()
+                            + "' instance is not available in "
+                            + VaadinContext.class.getName());
+        }
+        ClassLoader classloader = classLoaderAccess.getClassloader();
+
+        if (classloader instanceof BundleReference) {
+            return ((BundleReference) classloader).getBundle();
+        } else {
+            throw new IllegalStateException(
+                    "Unexpected classloader for the web app '" + classloader
+                            + "'. It's not possible to get an OSGi bundle from it");
+        }
     }
 }

--- a/flow-osgi/src/main/java/com/vaadin/flow/osgi/support/OSGiVaadinInitialization.java
+++ b/flow-osgi/src/main/java/com/vaadin/flow/osgi/support/OSGiVaadinInitialization.java
@@ -162,7 +162,7 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
 
         private final Bundle webAppBundle;
 
-        private final VaadinServletContext context;
+        private final VaadinServletContext servletContext;
 
         private final ServletContainerInitializerClasses initializerClasses;
 
@@ -173,7 +173,7 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
             super(webAppBundle.getBundleContext(),
                     ApplicationConfigurationFactory.class, null);
             this.webAppBundle = webAppBundle;
-            this.context = context;
+            servletContext = context;
             this.initializerClasses = initializerClasses;
         }
 
@@ -182,11 +182,11 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
                 ServiceReference<ApplicationConfigurationFactory> reference) {
             ApplicationConfigurationFactory factory = super.addingService(
                     reference);
-            AppConfigFactoryTracker tracker = context
+            AppConfigFactoryTracker tracker = servletContext
                     .getAttribute(AppConfigFactoryTracker.class);
             if (tracker != null) {
                 tracker.close();
-                context.removeAttribute(AppConfigFactoryTracker.class);
+                servletContext.removeAttribute(AppConfigFactoryTracker.class);
             }
             initializeLookup();
             return factory;
@@ -217,7 +217,7 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
             // called and the classloader should be available
 
             // ensure the lookup is set into the context
-            Lookup lookup = context.getAttribute(Lookup.class,
+            Lookup lookup = servletContext.getAttribute(Lookup.class,
                     this::createLookup);
             if (lookup == null) {
                 throw new IllegalStateException(
@@ -237,7 +237,7 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
                 }
             }
 
-            initializerClasses.addContext(context.getContext());
+            initializerClasses.addContext(servletContext.getContext());
         }
 
         private void handleUninitializedServlet(Lookup lookup,
@@ -258,7 +258,7 @@ public class OSGiVaadinInitialization implements VaadinServiceInitListener,
         }
 
         private Lookup createLookup() {
-            return new OsgiLookupImpl(webAppBundle, context);
+            return new OsgiLookupImpl(webAppBundle, servletContext);
         }
 
         private boolean isUninitializedServlet(Object object) {

--- a/flow-osgi/src/main/java/com/vaadin/flow/osgi/support/ServletContainerInitializerClasses.java
+++ b/flow-osgi/src/main/java/com/vaadin/flow/osgi/support/ServletContainerInitializerClasses.java
@@ -42,7 +42,7 @@ import com.vaadin.flow.router.HasErrorParameter;
 import com.vaadin.flow.server.InvalidApplicationConfigurationException;
 import com.vaadin.flow.server.startup.ClassLoaderAwareServletContainerInitializer;
 import com.vaadin.flow.server.startup.DevModeInitializer;
-import com.vaadin.flow.server.startup.LookupInitializer;
+import com.vaadin.flow.server.startup.LookupServletContainerInitializer;
 
 /**
  * Manages scanned classes inside OSGi container.
@@ -173,9 +173,9 @@ public final class ServletContainerInitializerClasses {
          * 
          * Lookup is set immediately in the context, so no need to initialize it
          */
-        initializerClasses.get().stream()
-                .filter(clazz -> !clazz.equals(DevModeInitializer.class)
-                        && !clazz.equals(LookupInitializer.class))
+        initializerClasses.get().stream().filter(
+                clazz -> !clazz.equals(DevModeInitializer.class) && !clazz
+                        .equals(LookupServletContainerInitializer.class))
                 .map(ReflectTools::createInstance)
                 .forEach(initializer -> handleTypes(initializer, context));
     }

